lib/smart_properties/property.rb

```diff
@@ -72,7 +72,11 @@ def convert(scope, value)
     end
 
     def default(scope)
-      @default.kind_of?(Proc) ? scope.instance_exec(&@default) : @default.dup
+      return scope.instance_exec(&@default) if @default.kind_of?(Proc)
+      return @default.dup unless @default.respond_to?(:duplicable?)
```

Comment:
Why not simply check for `dup`?

Reply:
I should have read the explanation in the description better. I'm against introducing any Rails specific code into a library that is otherwise fully Rails agnostic. The way I'd probably implement this feature is as follows:

```ruby
def default(scope)
  case @default
  when Proc
    scope.instance_exec(&@default)
  when Symbol
    @default
  else
    @default.dup
end
```

I like the idea of being specific about the things we don't duplicate because it is the desired default behaviour and not duplicating the default value has lead to issues in the past.